tests/rptest/tests/broken_connection_test.py

```diff
@@ -0,0 +1,83 @@
+# Copyright 2022 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+import time
+import random
+from rptest.services.cluster import cluster
+from rptest.services.redpanda import RESTART_LOG_ALLOW_LIST
+from rptest.clients.types import TopicSpec
+from rptest.tests.end_to_end import EndToEndTest
+from ducktape.mark import matrix
+
+
+class BrokenConnectionTest(EndToEndTest):
+    def __init__(self, ctx, *args, **kwargs):
+        super(BrokenConnectionTest, self).__init__(
+            ctx,
+            *args,
+            extra_rp_conf={
+                'enable_leader_balancer': False
+            },
+            **kwargs)
+        self._ctx = ctx
+
+    def _node_ip(self, node):
+        out = self.redpanda.nodes[0].account.ssh_output(f"getent hosts {node.name}")
+        return out.split()[0].decode()
+
+    def _node_ips(self):
+        return [(node, self._node_ip(node)) for node in self.redpanda.nodes]
+
+    def _make_rule(self, ip):
+        reject_types = ["icmp-host-unreachable", "tcp-reset"]
+        chains = ["INPUT", "OUTPUT"]
+        rule_targets = ["DROP", "REJECT"]
+
+        chain = random.choice(chains)
+        reject_type = random.choice(reject_types)
+        rule_target = random.choice(rule_targets)
+
+        rule = f"{chain} -s {ip} -p tcp --dport 33145 "
+        if rule_target == "DROP":
```

Comment:
we could level up this faults to some known set. 

`DROP_5PERCENT_PACKETS`-style enums/functions `drop_5percent_packets(src, dst)` `block_tcp(port, src, dst)` and then before the tests starts we print the faults that will be injected in the test. 

something like a class 


```python
class NetworkNemesis
  def init(src,dst)
  def apply() pass


class 5PercentDropNN: NetworkNemesis
    def __str__(self): 
       return "NetworkNemesis: Drop 5 Percent Packets"
    def apply(port)     
        __super__.apply() # whatever this is
        
       
TestWith(5PercentDropNN(src, dst),
                 BlockTCP(33145, src,dst))

```

Reply:
tl;dr is you can launch a test with a schedule of faults.

Reply:
nice, will do. i didn't consider adding faults like dropping packets because we generally assume we have a functioning tcp stack where as dropping connections and delaying transmissions affects the application directly. but i think we can expand this to be more general with a network nemsis. good idea